### PR TITLE
Add Fulcio certificate acceptance logic

### PR DESCRIPTION
**Pull request by mtrmac**
This is a, hopefully last, subset of #1778 , filing for a separate review.

@vrothberg PTAL

### CODE REVIEW

File: signature/fixtures/fulcio-chain [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+MTEwMDcxMzU2NTlaFw0zMTEwMDUxMzU2NThaMCoxFTATBgNVBAoTDHNpZ3N0b3Jl
+LmRldjERMA8GA1UEAxMIc2lnc3RvcmUwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAT7
+XeFT4rb3PQGwS4IajtLk3/OlnpgangaBclYpsYBr5i+4ynB07ceb3LP0OIOZdxex
+X69c5iVuyJRQ+Hz05yi+UF3uBWAlHpiS5sh0+H2GHE7SXrk1EC5m1Tr19L9gg92j
+YzBhMA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBRY
+wB5fkUWlZql6zJChkyLQKsXF+jAfBgNVHSMEGDAWgBRYwB5fkUWlZql6zJChkyLQ
+KsXF+jAKBggqhkjOPQQDAwNpADBmAjEAj1nHeXZp+13NWBNa+EDsDP8G1WWg1tCM
+WP/WHPqpaVo0jhsweNFZgSs0eE7wYI4qAjEA2WB9ot98sIkoF3vZYdd3/VtWB5b9
+TNMea7Ix/stJ5TfcLLeABLE4BNJOsQ4vnBHJ
+-----END CERTIFICATE-----
```
Comment (TomSweeneyRedHat) [inline signature/fixtures/fulcio-chain:27]: This is public, not private right?  I know pem files generally are and I'm assuming the files in here are too, but I don't know about this one.

Comment (TomSweeneyRedHat) [reply signature/fixtures/fulcio-chain:27]: and it looks like it's only being used by the tests, so probably a no-op

**Review by vrothberg — COMMENTED** [review]
Code LGTM, CI seems sick

**Comment by mtrmac** [comment]
@cevich It seems that Skopeo’s recent CI changes don’t work when invoked from c/image: `skopeo/systemtest/helpers.bash` is not available here.

**Comment by mtrmac** [comment]
The unit test failures are unexpected, adding a bit more detail to them…

**Comment by mtrmac** [comment]
The test unit failures are because unit tests run with Go 1.18.4, which is missing https://github.com/golang/go/commit/a11a885cb567b3797e33733e883c2ba3bdc0e898 .

@cevich Will the F37 update in #1799 update to Go ≥ 1.19, or should I change something about the implementation?

Maybe I should change the implementation anyway…

**Comment by mtrmac** [comment]
> Maybe I should change the implementation anyway…

Updated to explicitly reject duplicate issuers; we can’t assume Go 1.19 in all c/image users.

**Comment by mtrmac** [comment]
Tests pass now.

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM and very nice work @mtrmac 
Happy green test buttons

**Comment by cevich** [comment]
> Will the F37 update in https://github.com/containers/image/pull/1799 update to Go ≥ 1.19

Yes, 1.19 and going forward.

**Review by vrothberg — APPROVED** [review]
LGTM

**Pull request merged**